src/mbgl/style/expression/number_format.cpp

```diff
@@ -0,0 +1,291 @@
+#include <mbgl/style/expression/number_format.hpp>
+#include <mbgl/style/conversion_impl.hpp>
+
+#include <locale>
+#include <sstream>
+#include <iomanip>
+
+namespace mbgl {
+namespace style {
+namespace expression {
+
+NumberFormat::NumberFormat(std::unique_ptr<Expression> number_,
+                           optional<std::unique_ptr<Expression>> locale_,
+                           optional<std::unique_ptr<Expression>> currency_,
+                           optional<std::unique_ptr<Expression>> minFractionDigits_,
+                           optional<std::unique_ptr<Expression>> maxFractionDigits_)
+    : Expression(Kind::NumberFormat, type::String),
+      number(std::move(number_))
+{
+      if (locale_) {
+            locale = std::unique_ptr<Expression>(std::move(*locale_));
+      }
+      if (currency_) {
+            currency = std::unique_ptr<Expression>(std::move(*currency_));
```

Comment:
`currency = std::move(*currency_);` , but what if `currency_` had been initialized with with `std::unique_ptr<Expression>()` (aka `nullptr`) ? Looks like we do not check it.